napari/utils/colormaps/standardize_color.py

```diff
@@ -0,0 +1,331 @@
+"""This module contains functions that 'standardize' the color handling
+of napari layers by supplying functions that are able to convert most
+color representation the user had in mind into a single representation -
+a numpy Nx4 array of float32 values - that is used across the codebase.
+
+The main function of the module is "transform_color", which might call
+a cascade of other, private, function in the module to do the hard work
+of converting the input.
+
+In general, we try to catch invalid color representations, warn the users
+of their misbehaving and return a default white color array, since it seems
+unreasonable to crash the entire napari session to mis-represented colors.
+"""
+
+import types
+from typing import Iterable, Union, Dict, Any, Callable
+import functools
+import warnings
+
+import numpy as np
+from vispy.color import Color, ColorArray, get_color_dict, get_color_names
+from vispy.color.color_array import _string_to_rgb
+
+
+def transform_color(colors: Any) -> np.ndarray:
+    """Receives the user-given colors and transforms them into an array of
+    Nx4 np.float32 values, with N being the number of colors. The function
+    (via its subfunctions, marked with _handle_X) is designed to parse all
+    valid color representations a user might have and convert them properly.
+    That being said, combinations of different color representation in the
+    same list of colors is prohibited, and will error. This means that a list
+    of ['red', np.array([1, 0, 0])] cannot be parsed and has to be manually
+    pre-processed by the user before sent to this function.
+
+    Parameters
+    --------
+    colors : string, array-like, Color and ColorArray instances, or a mix of the above.
+        The color(s) to interpret and convert
+
+    Returns
+    ------
+    colors : np.ndarray
+        An instance of np.ndarray with a data type of float32, 4 columns in
+        RGBA order and N rows, with N being the number of colors. The array
+        will always be 2D even if a single color is passed.
+
+    Raises
+    -----
+    ValueError, AttributeError, KeyError
+        invalid inputs
+    """
+    colortype = type(colors)
+    return _color_switch[colortype](colors)
```

Comment:
Very fancy! :joy: I like it! Always thought Python could use a switch statement. =D